diff --git a/node_modules/@projectdysnomia/dysnomia/lib/voice/VoiceConnection.js b/node_modules/@projectdysnomia/dysnomia/lib/voice/VoiceConnection.js
index 5b33fb4..9a76f47 100644
--- a/node_modules/@projectdysnomia/dysnomia/lib/voice/VoiceConnection.js
+++ b/node_modules/@projectdysnomia/dysnomia/lib/voice/VoiceConnection.js
@@ -422,6 +422,7 @@ class VoiceConnection extends EventEmitter {
                     this.connecting = false;
                     this.reconnecting = false;
                     this.ready = true;
+                    if(this.paused) this.resume();
                     /**
                      * Fired when the voice connection was resumed
                      * @event VoiceConnection#resumed
@@ -572,7 +573,14 @@ class VoiceConnection extends EventEmitter {
             this.emit("warn", `Voice WS close ${code}: ${reason}`);
             if(this.connecting || this.ready) {
                 let reconnecting = true;
-                if(code === 1000 || code === 4006 || code === 4021) {
+                if(code === 4006) {
+                    if (this.channelID) {
+                        reconnecting = true;
+                        err = null;
+                    } else {
+                        reconnecting = false;
+                    }
+                } else if(code === 1000 || code === 4021) {
                     reconnecting = false;
                 } else if(code === 4014 || code === 4022) {
                     if(this.channelID) {
@@ -584,11 +592,11 @@ class VoiceConnection extends EventEmitter {
                     }
                 }
 
-                this.disconnect(err, reconnecting);
+                const canResume = code === 4015 || code < 4000;
+                this.disconnect(err, reconnecting, canResume);
 
                 if(reconnecting) {
                     // Reset resume params if we cant resume
-                    const canResume = code === 4015 || code < 4000;
                     if(!canResume) {
                         this.wsSequence = -1;
                         this.sessionID = null;
@@ -680,7 +688,7 @@ class VoiceConnection extends EventEmitter {
         return this.daveSession?.voicePrivacyCode ?? undefined;
     }
 
-    disconnect(error, reconnecting) {
+    disconnect(error, reconnecting, resume = false) {
         this.connecting = false;
         this.reconnecting = reconnecting;
         this.ready = false;
@@ -691,11 +699,15 @@ class VoiceConnection extends EventEmitter {
         }
 
         try {
-            if(!reconnecting) {
+            if(!resume) {
                 this.speaking = false;
                 this.timestamp = 0;
                 this.sequence = 0;
-                this.stopPlaying();
+                if(reconnecting) {
+                    this.pause();
+                } else {
+                    this.stopPlaying();
+                }
             }
         } catch(err) {
             this.emit("error", err);
@@ -704,7 +716,7 @@ class VoiceConnection extends EventEmitter {
             clearInterval(this.heartbeatInterval);
             this.heartbeatInterval = null;
         }
-        if(!reconnecting) {
+        if(!resume) {
             this.#closeUDP();
         }
         if(this.ws) {
diff --git a/node_modules/@projectdysnomia/dysnomia/lib/voice/streams/OggOpusTransformer.js b/node_modules/@projectdysnomia/dysnomia/lib/voice/streams/OggOpusTransformer.js
index fc4016a..cfc778c 100644
--- a/node_modules/@projectdysnomia/dysnomia/lib/voice/streams/OggOpusTransformer.js
+++ b/node_modules/@projectdysnomia/dysnomia/lib/voice/streams/OggOpusTransformer.js
@@ -5,6 +5,7 @@ const BaseTransformer = require("./BaseTransformer");
 class OggOpusTransformer extends BaseTransformer {
     #bitstream = null;
     #remainder = null;
+    #incompleteSegment = null;
 
     process(buffer) {
         if(buffer.length - buffer._index <= 26) {
@@ -16,9 +17,6 @@ class OggOpusTransformer extends BaseTransformer {
         }
 
         const typeFlag = buffer.readUInt8(buffer._index + 5);
-        if(typeFlag === 1) {
-            return new Error("OGG continued page not supported");
-        }
 
         const bitstream = buffer.readUInt32BE(buffer._index + 14);
 
@@ -58,6 +56,10 @@ class OggOpusTransformer extends BaseTransformer {
                 if(byte === "OpusTags") {
                     this.emit("debug", segment.toString());
                 } else if(bitstream === this.#bitstream) {
+                    if (typeFlag === 1 && this.#incompleteSegment) {
+                        segment = Buffer.concat([this.#incompleteSegment, segment]);
+                        this.#incompleteSegment = null;
+                    }
                     this.push(segment);
                 }
             } else if(byte === "OpusHead") {
@@ -67,6 +69,12 @@ class OggOpusTransformer extends BaseTransformer {
                 this.emit("debug", "Invalid codec: " + byte);
             }
         }
+
+        // Save incomplete segment
+        if (size !== 0) {
+            buffer._index += size;
+            this.#incompleteSegment = buffer.subarray(buffer._index - size, buffer._index);
+        }
     }
 
     _final(cb) {
