diff --git a/node_modules/eris/lib/voice/VoiceConnection.js b/node_modules/eris/lib/voice/VoiceConnection.js
index 6c0c24b..fdacf9a 100644
--- a/node_modules/eris/lib/voice/VoiceConnection.js
+++ b/node_modules/eris/lib/voice/VoiceConnection.js
@@ -337,7 +337,7 @@ class VoiceConnection extends EventEmitter {
                 if(code === 4006) {
                     reconnecting = false;
                 } else if(code === 4014) {
-                    reconnecting = false;
+                    // reconnecting = false;
                 } else if(code === 1000) {
                     reconnecting = false;
                 }
