diff --git a/node_modules/eris/lib/voice/VoiceConnection.js b/node_modules/eris/lib/voice/VoiceConnection.js
index 0c8ff0e..6c3f707 100644
--- a/node_modules/eris/lib/voice/VoiceConnection.js
+++ b/node_modules/eris/lib/voice/VoiceConnection.js
@@ -120,6 +120,7 @@ class VoiceConnection extends EventEmitter {
         this.connectionTimeout = null;
         this.connecting = false;
         this.reconnecting = false;
+        this.resumeing = false;
         this.ready = false;
 
         this.sendBuffer = Buffer.allocUnsafe(16 + 32 + MAX_FRAME_SIZE);
@@ -199,6 +200,14 @@ class VoiceConnection extends EventEmitter {
                 clearTimeout(this.connectionTimeout);
                 this.connectionTimeout = null;
             }
+            if(this.resumeing) {
+                this.sendWS(VoiceOPCodes.RESUME, {
+                    server_id: this.id === "call" ? data.channel_id : this.id,
+                    session_id: data.session_id,
+                    token: data.token
+                })
+                return;
+            }
             this.sendWS(VoiceOPCodes.IDENTIFY, {
                 server_id: this.id === "call" ? data.channel_id : this.id,
                 user_id: data.user_id,
@@ -269,6 +278,11 @@ class VoiceConnection extends EventEmitter {
                     this.sendUDPPacket(udpMessage);
                     break;
                 }
+                case VoiceOPCodes.RESUMED: {
+                    this.connecting = false;
+                    this.resumeing = false;
+                    break;
+                }
                 case VoiceOPCodes.SESSION_DESCRIPTION: {
                     this.mode = packet.d.mode;
                     this.secret = Buffer.from(packet.d.secret_key);
@@ -355,8 +369,20 @@ class VoiceConnection extends EventEmitter {
             this.emit("warn", `Voice WS close ${code}: ${reason}`);
             if(this.connecting || this.ready) {
                 let reconnecting = true;
+                if(code < 4000 || code === 4015) {
+                    this.resumeing = true;
+                    setTimeout(() => {
+                        this.connect(data);
+                    }, 500).unref();
+                    return;
+                }
                 if(code === 4006) {
-                    reconnecting = false;
+                    if(this.channelID) {
+                        reconnecting = true;
+                        err = null;
+                    } else {
+                        reconnecting = false;
+                    }
                 } else if(code === 4014) {
                     if(this.channelID) {
                         data.endpoint = null;
@@ -383,6 +409,7 @@ class VoiceConnection extends EventEmitter {
     disconnect(error, reconnecting) {
         this.connecting = false;
         this.reconnecting = reconnecting;
+        this.resumeing = false;
         this.ready = false;
         this.speaking = false;
         this.timestamp = 0;
diff --git a/node_modules/eris/lib/voice/streams/BaseTransformer.js b/node_modules/eris/lib/voice/streams/BaseTransformer.js
index 7160c6c..4d4917c 100644
--- a/node_modules/eris/lib/voice/streams/BaseTransformer.js
+++ b/node_modules/eris/lib/voice/streams/BaseTransformer.js
@@ -10,7 +10,7 @@ class BaseTransformer extends TransformStream {
             options.allowHalfOpen = true;
         }
         if(options.highWaterMark === undefined) {
-            options.highWaterMark = 0;
+            options.highWaterMark = 1;
         }
         super(options);
         this.manualCB = false;
diff --git a/node_modules/eris/lib/voice/streams/OggOpusTransformer.js b/node_modules/eris/lib/voice/streams/OggOpusTransformer.js
index 6c4baed..204a05e 100644
--- a/node_modules/eris/lib/voice/streams/OggOpusTransformer.js
+++ b/node_modules/eris/lib/voice/streams/OggOpusTransformer.js
@@ -8,6 +8,7 @@ class OggOpusTransformer extends BaseTransformer {
 
         this._remainder = null;
         this._bitstream = null;
+        this._incompleteSegment = null;
     }
 
     process(buffer) {
@@ -20,10 +21,6 @@ class OggOpusTransformer extends BaseTransformer {
         }
 
         const typeFlag = buffer.readUInt8(buffer._index + 5);
-        if(typeFlag === 1) {
-            return new Error("OGG continued page not supported");
-        }
-
         const bitstream = buffer.readUInt32BE(buffer._index + 14);
 
         buffer._index += 26;
@@ -62,6 +59,10 @@ class OggOpusTransformer extends BaseTransformer {
                 if(byte === "OpusTags") {
                     this.emit("debug", segment.toString());
                 } else if(bitstream === this._bitstream) {
+                    if (typeFlag === 1 && this._incompleteSegment) {
+                        segment = Buffer.concat([this._incompleteSegment, segment]);
+                        this._incompleteSegment = null;
+                    }
                     this.push(segment);
                 }
             } else if(byte === "OpusHead") {
@@ -71,6 +72,12 @@ class OggOpusTransformer extends BaseTransformer {
                 this.emit("debug", "Invalid codec: " + byte);
             }
         }
+
+        // Save incomplete segment
+        if (size !== 0) {
+            buffer._index += size;
+            this._incompleteSegment = buffer.subarray(buffer._index - size, buffer._index);
+        }
     }
 
     _final() {
